### PR TITLE
Support kernels in unsaved files

### PR DESCRIPTION
**Pull request by BenRussert**
@kylebarron and I have been collaborating on this. 

The branch is ready for a spin if you want to try it. My plan is just to let this bake a few days in case of comments/thoughts/ideas/issues we think of.

### CODE REVIEW

File: lib/store/index.js [hunk 1]
```diff
@@ -39,23 +39,30 @@ export class Store {
...
       const currentScopeName = this.grammar.scopeName;
-      return this.runningKernels.filter(
+      return this.runningKernels.find(
         k => k.grammar.scopeName === currentScopeName
-      )[0];
+      );
     }
-
-    if (!this.filePath) return null;
+    //
```
Comment (kylebarron) [inline lib/store/index.js]: Empty comment?

File: lib/store/index.js [hunk 2]
```diff
@@ -174,28 +181,32 @@ export class Store {
...
+
+    const embeddedScope = getEmbeddedScope(
+      editor,
+      editor.getCursorBufferPosition()
+    );
+
+    if (!embeddedScope) return grammar;
+    const scope = embeddedScope.replace(".embedded", "");
+    const grammarForScopeName = atom.grammars.grammarForScopeName(scope);
+    return atom.grammars.grammarForScopeName(scope);
```
Comment (kylebarron) [inline lib/store/index.js]: Is `grammarForScopeName` used?

File: lib/store/index.js (hunk 1 above)
Comment (BenRussert) [reply lib/store/index.js]: Invisible comment 🧙‍♂️. We can remove it with magic.

File: lib/store/index.js (hunk 2 above)
Comment (BenRussert) [reply lib/store/index.js]: Nope, good catch. We could drop the line above or use the variable name.

Comment (kylebarron) [reply lib/store/index.js]: either is fine

**Comment by BenRussert** [comment]
@kylebarron you can push those changes if you want and merge.

**Comment by BenRussert** [comment]
Great to have this back, thanks Kyle!

**Pull request merged**